src/util/network_interface.go

```
@@ -47,6 +48,21 @@ func (n *NetworkInterface) Addrs() ([]net.Addr, error) {
 	return n.netInterface.Addrs()
 }
 
+func (n *NetworkInterface) Type() string {
+	evaledPath, err := n.dependencies.EvalSymlinks(fmt.Sprintf("/sys/class/net/%s", n.netInterface.Name))
+	if err != nil {
+		logrus.WithError(err).Warnf("Could not determine if interface %s is physical", n.netInterface.Name)
+		return err.Error()
+	} else if !strings.Contains(evaledPath, "/virtual/") {
+		return "physical"
+	}
+	link, err := n.dependencies.LinkByName(n.netInterface.Name)
+	if err != nil {
+		return err.Error()
```

If you can return an error, then this function should return an error type.

Doing it this way would lead to us putting the error string as the interface type in inventory, but that's definitely not what we want, right?

---

That's right! Thanks for catching that. Modified it!